### PR TITLE
feat(scrapers): support nytrial scrapers

### PR DESCRIPTION
**Pull request by grossir**
Theses changes are needed to support freelawproject/juriscraper#827

- If possible, get court object from child_court field passed by nytrial families of scrapers. Otherwise, default to parent court. This does not alter behavior for other sources. Solves freelawproject/juriscraper#847
- Pass opinion.html to site.extract_from_text, if opinion.plain_text does not exists. Solves #3549
- Add support to update Opinion object from extract_from_text metadata dict.

### CODE REVIEW

File: cl/scrapers/management/commands/cl_scrape_opinions.py [hunk 1]
```diff
@@ -238,6 +239,23 @@ def scrape_court(self, site, full_crawl=False, ocr_available=True):
         logger.debug(f"#{len(site)} opinions found.")
         added = 0
         for i, item in enumerate(site):
+            # Get court id from child court string passed by nytrial scrapers
+            child_court = None
+            if court_str.startswith("ny") and item.get("child_courts"):
+                child_court_id = find_court_ids_by_name(
+                    item["child_courts"], False, "New York", False
+                )
```
Comment (flooie) [inline cl/scrapers/management/commands/cl_scrape_opinions.py:247]: I wouldnt mind seeing the parameters names

File: cl/scrapers/tasks.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def update_document_from_text(opinion: Opinion) -> None:
     site = get_scraper_object_by_name(court)
     if site is None:
         return
-    metadata_dict = site.extract_from_text(opinion.plain_text)
+    metadata_dict = site.extract_from_text(opinion.plain_text or opinion.html)
```
Comment (flooie) [inline cl/scrapers/tasks.py:60]: Thanks for fixing this.

**Comment by flooie** [comment]
I think you need to include the latest release of courts-db v0.10.21 to poetry. Otherwise I think this is complete

**Review by flooie — CHANGES_REQUESTED** [review]
Update courts-db to v0.10.21 please.

**Comment by grossir** [comment]
I abstracted the behavior for child courts into get_child_court
I tested it locally creating the following courts:
```
from cl.search.models import Court

# Child court with expected parent
Court(id='nysupctnewyork', parent_court=Court.objects.get(pk='nysupct'), position='33.1', jurisdiction='TRT', full_name='New York County, Supreme Court', short_name='Test2').save()

# Child courts with unexpected parents, error should be logged and records saved with court id from scraper
Court(id='nysupctqueens', parent_court=Court.objects.get(pk='nysurct'), position='35.2', jurisdiction='TRT', full_name='Queens County, Supreme Court', short_name='Test3').save()
Court(id='nysupctkings', parent_court=Court.objects.get(pk='nysurct'), position='36.2', jurisdiction='TRT', full_name='Kings County, Supreme Court', short_name='Test3').save()
```
and checking that both logs and DB behaved as expected, and they did

File: cl/scrapers/utils.py [hunk 3]
```diff
@@ -20,7 +21,52 @@
 from cl.lib.microservice_utils import microservice
 from cl.recap.mergers import find_docket_object
 from cl.scrapers.tasks import extract_recap_pdf
-from cl.search.models import Docket, RECAPDocument
+from cl.search.models import Court, Docket, RECAPDocument
+
+
+def get_child_court(child_court_name: str, court: Court) -> Optional[Court]:
```
Comment (flooie) [inline cl/scrapers/utils.py:27]: I would not pass the full court object when the string will do.

File: cl/scrapers/management/commands/cl_scrape_opinions.py [hunk 4]
```diff
@@ -302,8 +304,10 @@ def scrape_court(self, site, full_crawl=False, ocr_available=True):
             )
             dup_checker.reset()
 
+            child_court = get_child_court(item.get("child_courts", ""), court)
```
Comment (flooie) [inline cl/scrapers/management/commands/cl_scrape_opinions.py:307]: something about this is weird to me ...

**Review by flooie — APPROVED** [review]
Thanks @grossir  for putting this through its paces.

**Comment by flooie** [comment]
@grossir what say you?

**Comment by grossir** [comment]
From the Sentry event, the child court name was  `'Ct Cl'` and the child court id was `nyclaimsct`
So, it found the "parent" court as child. 
I think this merits a new control in that function, to ensure that the child found actually has a parent.

**Pull request merged**